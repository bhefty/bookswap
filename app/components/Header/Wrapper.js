import styled from 'styled-components'

export default styled.div`
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1101;

  header {
    height: 48px;

    img {
      padding-bottom: 5px;
      height: 100%;
      margin-right: 1em;
    }

    .pull-right {
      float: right;
    }
  }
`
