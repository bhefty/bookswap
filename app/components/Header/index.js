import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'

import Img from 'components/Img'
import logo from 'assets/logo_white.png'
import Wrapper from './Wrapper'

class Header extends PureComponent {
  render () {
    return (
      <Wrapper>
        <header className='sticky'>
          <label className='drawer-toggle button' htmlFor='navigation-toggle' />
          <Link to='/' className='logo hidden-sm'>
            <Img src={logo} alt='React Redux Boilerplate Logo' />
            <span>React Redux Boilerplate</span>
          </Link>
          {!this.props.profile
            ? // eslint-disable-line operator-linebreak
              <button className='button pull-right' onClick={this.props.handleLogin}>Login</button>
            : // eslint-disable-line operator-linebreak
              <button className='button pull-right' onClick={this.props.handleLogout}>Logout</button>
          }
          <Link to='/features' className='button pull-right hidden-sm'>Features</Link>
          {this.props.profile &&
            <Link to='/dashboard' className='button pull-right hidden-sm'>Dashboard</Link>
          }
        </header>
        <input id='navigation-toggle' type='checkbox' />
        <nav className='drawer hidden-md hidden-lg'>
          <label htmlFor='navigation-toggle' className='close' />
          <br />
          <Link to='/'>Home</Link>
          <Link to='/features'>Features</Link>
        </nav>
      </Wrapper>
    )
  }
}

export default Header
