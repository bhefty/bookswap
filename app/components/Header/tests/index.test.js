import React from 'react'
import { shallow } from 'enzyme'

import Header from '../index'

describe('<Header />', () => {
  it('should render a header', () => {
    const renderedComponent = shallow(
      <Header />
    )
    expect(renderedComponent.find('header').length).toEqual(1)
  })

  it('should render Dashboard link if logged in', () => {
    const profile = {
      email: 'joeschmoe@example.com',
      name: 'Joe Schmoe'
    }
    const renderedComponent = shallow(
      <Header profile={profile} />
    )
    expect(renderedComponent.contains('Dashboard')).toBe(true)
  })

  it('should render Logout button if logged in', () => {
    const profile = {
      email: 'joeschmoe@example.com',
      name: 'Joe Schmoe'
    }
    const renderedComponent = shallow(
      <Header profile={profile} />
    )
    expect(renderedComponent.contains('Logout')).toBe(true)
  })

  it('should render Login button if not logged in', () => {
    const renderedComponent = shallow(
      <Header />
    )
    expect(renderedComponent.contains('Login')).toBe(true)
  })
})
