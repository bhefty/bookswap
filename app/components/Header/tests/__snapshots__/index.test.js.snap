// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the page 1`] = `
<header>
  <h1>
    Simple React Page
  </h1>
</header>
`;
